### ISSUE TITLE
Docker hangs and unable to reboot

### ISSUE THREAD

**Issue by chanderson0** [coreos/bugs#1998]
### Container Linux Version ###

```
core@staging ~ $  cat /etc/os-release
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=1409.1.0
VERSION_ID=1409.1.0
BUILD_ID=2017-05-23-2230
PRETTY_NAME="Container Linux by CoreOS 1409.1.0 (Ladybug)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
```

```
core@staging ~ $ docker info
Containers: 12
 Running: 12
 Paused: 0
 Stopped: 0
Images: 8
Server Version: 1.12.6
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp selinux
Kernel Version: 4.11.2-coreos
Operating System: Container Linux by CoreOS 1409.1.0 (Ladybug)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 14.69 GiB
Name: staging.hostname.com
ID: VHNN:RTQD:WF6Q:LA4J:AUSK:CM2Q:TJCR:5N3B:ICJD:3Q4C:TE6N:2GNG
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

### Environment ###

What hardware/cloud provider/hypervisor is being used to run Container Linux?
AWS EC2 ami-773e7417

### Expected Behavior ###

Docker can start/stop containers normally.

### Actual Behavior ###

Docker commands hang (ps, kill, etc.). Cannot restart the docker process with systemctl. Cannot reboot the system except by stopping / starting the machine through AWS. This happens not immediately but after a couple of days of running.

Notably, we start seeing messages of the form: `kernel: unregister_netdevice: waiting for veth9e99380 to become free. Usage count = -1` for a while, then it seems like those messages block the machine from rebooting normally.

Logs from `kill SIGUSR1` before reboot and `journalctl -r -b -1` after hard reboot: https://gist.github.com/chanderson0/f837b40e1d62562922b4ef85586178aa

### Reproduction Steps ###

This happens consistently across our staging and development environments - on CoreOS stable and beta. I'm trying alpha now to see if anything there helps.

### Other Information ###

The only other notable things about our setup are: 
- we start and stop containers quickly (every few seconds) to handle short-running processes
- we do a little custom networking setup to use ipset to filter ips (thought I'd mention since it's maybe networking related): https://gist.github.com/chanderson0/fd8872f324e1358d0ea45cf52e10263f

**Comment by squeed** [comment]
It looks like this is a (long-standing issue)[moby/moby#5618] with docker.

**Comment by chanderson0** [comment]
I thought so too, but I have yet to see someone report negative values for "usage count", so posted a new issue.

**Comment by lucab** [comment]
Yes, this is similar to the upstream ticket (also already reported at https://github.com/coreos/bugs/issues/254), but the negative counter is a more interesting twist pointing this toward a different root cause. It is still a kernel bug/race, but this time some module is releasing the interface too many times.

It would interesting to know:
 * what is the lowest kernel version you experienced this
 * if you manage to reproduce this with a similar setup but without the ipset bits (i.e. simplifying the iptables rules to something simpler without ipset)
 * if kernel logs contain some useful entries when the counter goes negative

**Comment by squeed** [comment]
Hmm, interesting. It looks like some people have combined the positive and negative usage counts. I think you're right - these are probably different bugs.

**Comment by chanderson0** [comment]
- I can't verify the lowest version, but it's currently happening on CoreOS beta channel which reports (from `uname -r`) `4.11.2-coreos`.
- We had this issue months ago with a more complicated setup, but it seems to have gotten worse even as the iptables rules got simpler. We could try a lighter setup on another machine, but it'll take a bit to set that up.
- Good call on the kernel logs, I hadn't happened upon this yet. Have some interesting traces in the minutes leading up to the first negative number: https://gist.github.com/chanderson0/3078937de018ea7168704c0f6e2ce0d0

Last point: I switched another machine to CoreOS alpha channel and haven't seen this issue in 24h. I don't want to keep actual services running on alpha, but it's better than having to reboot every few hours...

**Comment by euank** [comment]
If alpha actually does fix it, it's probably due to changes in the docker daemon (1.12 vs 17.05) rather than the kernel.

The alpha kernel is 4.11.3 (vs 4.11.2 on beta), and the changelog for 4.11.3 has nothing that looks relevant: https://lkml.org/lkml/2017/5/25/493

**Comment by chanderson0** [comment]
Closing this - haven't seen the issue in 3 days on CoreOS alpha. Will re-open if it crops up again. Thanks!

**Comment by chanderson0** [comment]
Reopening - now seeing this on CoreOS alpha 1451.2.0 (docker 17.05.0 / 4.11.6) and beta 1437.3.0 (docker 1.12.6 / kernel 4.11.8).

Since I'm seeing a lot of errors that looks like `systemd-udevd[10478]: Could not generate persistent MAC address for veth7990983: No such file or directory`, I'm attempting the configuration change suggested here: https://github.com/systemd/systemd/issues/3374#issuecomment-288882355. I don't really expect that to change much, but at least it might make the logs cleaner...

Here are some more logs from the machine running the alpha: https://gist.github.com/chanderson0/1fb5d20b10d5903e517bfb40b534e4af

Anything other data I can supply or anything else you can suggest trying? Once the machines get into this state, docker can't be killed and that machine becomes unusable until a hard reboot. I'm probably going to have to switch to a non-CoreOS deployment pretty soon if I can't get this to work :(

**Comment by bgilbert** [comment]
Are you still seeing this with current Container Linux releases?  Stable 1688.5.3 has Docker 17.12.1 and kernel 4.14.32.
